src/gui/src/settings/options-window.cpp

```diff
@@ -342,6 +343,151 @@ OptionsWindow::OptionsWindow(Profile *profile, ThemeLoader *themeLoader, QWidget
 	ui->comboImagePositionVideoH->setCurrentIndex(positionsH.indexOf(settings->value("imagePositionVideoH", "left").toString()));
 	ui->lineImageBackgroundColor->setText(settings->value("imageBackgroundColor", QString()).toString());
 
+	ui->checkRememberDrawer->setChecked(settings->value("Zoom/rememberDrawer", true).toBool());
+	ui->checkRememberGeometry->setChecked(settings->value("Zoom/rememberGeometry", true).toBool());
+
+	if (settings->value("firstload", true).toBool()) {
+		// Fix for tristates, which don't seem to be supported by Designer's ui files.
+		ui->checkButtonSaveFav->setCheckState(Qt::PartiallyChecked);
+		ui->checkButtonSaveNQuitFav->setCheckState(Qt::PartiallyChecked);
+		ui->checkButtonOpenFav->setCheckState(Qt::PartiallyChecked);
+	} else {
+		log("+++Reading Zoom/Buttons+++");
+		settings->beginGroup("Zoom");
+		QList<ButtonSettings> buttons = settings->value("allButtons").value<QList<ButtonSettings>>();
+		for (QList<ButtonSettings>::iterator button = buttons.begin(); button != buttons.end(); button++) {
```

Comment:
You should really start using for-range loops when using C++, it's just much more elegant 😁
I'll push a style commit for this kind of stuff, so you can ignore this comment.

Reply:
They are more elegant. I have been trying to use them more since you mentioned it before but this code is from a while back. I do think there's some advantage to knowing what kind of object is being iterated just by looking at the loop though, that's why I don't have a strong preference personally.